service/matching/task_reader.go

```diff
@@ -54,14 +56,15 @@ type (
 		backlogMgr *backlogManagerImpl
 		gorogrp    goro.Group
 
-		backoffTimerLock sync.Mutex
-		backoffTimer     *time.Timer
-		retrier          backoff.Retrier
+		backoffTimerLock      sync.Mutex
+		backoffTimer          *time.Timer
+		retrier               backoff.Retrier
+		backlogHeadCreateTime atomic.Int64
```

Comment:
wait, TaskMatcher has backlog age tracking already.. why are we adding a second implementation? I remember discussing this but forgot the conclusion.

if this is to avoid the momentary zero problem, is there a way to fix the other tracking mechanism to avoid the momentary zero?

if not, could the other implementation be replaced with this one? or is the problem that they're actually tracking slightly different things? (source vs dest for redirections?)

if that's the case, I would like to see some comments (on both backlog age trackers) that clearly describe what the differences are and why there are two of them.

Reply:
- I had a discussion with @ShahabT about this and the main reasoning to include this new metric was to avoid the momentary zero problem. This is documented as a comment for the `matcher.getBacklogAge()` and reads:
```
// getBacklogAge is the latest age across all backlogs re-directing to this matcher; may momentarily
// be 0 cause of race conditions when no reader pushes a task into the matcher at this moment
```

- AFAIK, the easier (and best, i guess) solution to fix this is to come up with this second counter which does tracking by simply reading the backlog age of the task read by the reader. I also do believe that they are tracking different things.

@ShahabT , have I missed something or is this an accurate explanation of our discussion?